src/main.ts

```diff
@@ -148,212 +98,108 @@ export class Content {
-        owner: repo.owner,
-        repo: repo.name,
-        path: 'action.yaml'
-      })
-
-      if ('name' in yaml && 'download_url' in yaml) {
-        result.name = yaml.name
-        result.owner = repo.owner
-        result.repo = repo.name
-        result.forkedfrom = parentinfo
-        if (yaml.download_url !== null) {
-          result.downloadUrl = yaml.download_url
-        }
-      }
-    } catch (error) {
-      core.debug(`No action.yaml file found in repository: ${repo.name}`)
-    }
+  if (organization !== '') {
+    core.info(`Search for action files under the organization [ ${organization} ]`)
+    searchQuery = searchQuery.concat('+org:', organization)
   }
 
-  // todo: ratelimiting can be enabled on GHES as well, but is off by default
-  // we can probably load it from an api call and see if it is enabled, or try .. catch 
-  if (!isEnterpriseServer) {
-    // search API has a strict rate limit, prevent errors
-    var ratelimit = await client.rest.rateLimit.get()
-    if (ratelimit.data.resources.search.remaining <= 2) {
-      // show the reset time
-      var resetTime = new Date(ratelimit.data.resources.search.reset * 1000)
-      core.debug(`Search API reset time: ${resetTime}`)
-      // wait until the reset time
-      var waitTime = resetTime.getTime() - new Date().getTime()
-      if (waitTime < 0) {
-        // if the reset time is in the past, wait 2,5 seconds for good measure (Search API rate limit is 30 requests per minute)
-        waitTime = 2500
-      } else {
-        // back off a bit more to be more certain
-        waitTime = waitTime + 1000
-      }
-      core.info(`Waiting ${waitTime/1000} seconds to prevent the search API rate limit`)
-      await new Promise(r => setTimeout(r, waitTime));
-    }
-  }
-  if (!result.name) {
-    core.info(`No actions found at root level in repository: ${repo.name}`)
-    core.info(`Checking subdirectories in repository: ${repo.name}`)
-    var searchQuery = '+filename:action+language:YAML+repo:' + repo.owner + '/' + repo.name;
+  core.debug(`searchQuery : ${searchQuery}`)
 
-    var searchResultforRepository = await client.request("GET /search/code", {
-      q: searchQuery
-    });
+  const searchResult = await client.paginate(client.rest.search.code, {
+    q: searchQuery
+  })
 
-    if (Object.keys(searchResultforRepository.data.items).length > 0) {
+  if(searchResult){
+    for (let index = 0; index <searchResult.length; index++) {
+      const result = new Content()
+      var fileName = searchResult[index].name; 
+      var element = searchResult[index].path;
+      var repoName = searchResult[index].repository.name;
+      var repoOwner = searchResult[index].repository.owner.login;
+
+      // Push file to action list if filename matches action.yaml or action.yml
+      // Search result will contains list of files matching action files ex: reposyncer_action.yml
+      if(fileName == 'action.yaml'|| fileName == 'action.yml'){
```

Comment:
```suggestion
      if (fileName == 'action.yaml'|| fileName == 'action.yml'){
```
Follow layout of the rest of the code

Reply:
Updated